packages/core/src/utils/StorageUtil.ts

```diff
@@ -280,5 +294,238 @@ export const StorageUtil = {
     } catch {
       console.info('Unable to remove connected namespace')
     }
+  },
+  getBalanceCache() {
+    let cache: Record<string, { timestamp: number; balance: BlockchainApiBalanceResponse }> = {}
+    try {
+      const result = SafeLocalStorage.getItem(SafeLocalStorageKeys.BALANCE_CACHE)
+      cache = result ? JSON.parse(result) : {}
+    } catch {
+      console.info('Unable to get balance cache')
+    }
+
+    return cache
+  },
+  removeAddressFromBalanceCache(caipAddress: string) {
+    try {
+      const cache = StorageUtil.getBalanceCache()
+      SafeLocalStorage.setItem(
+        SafeLocalStorageKeys.BALANCE_CACHE,
+        JSON.stringify({ ...cache, [caipAddress]: undefined })
+      )
+    } catch {
+      console.info('Unable to remove address from balance cache', caipAddress)
+    }
+  },
+  getBalanceCacheForCaipAddress(caipAddress: string) {
+    try {
+      const cache = StorageUtil.getBalanceCache()
+      const balanceCache = cache[caipAddress]
+      // We want to discard cache if it's older than the cache expiry
+      if (balanceCache && Date.now() - balanceCache?.timestamp < this.cacheExpiry.balance) {
+        return balanceCache.balance
+      }
+
+      console.info('Discarding balanceCache cache for address', caipAddress)
```

Comment:
don't we need to do this for all cache handling? 
if so we shuold add a util like `isCacheExpired`